patches/tModLoader/Terraria/Framing.cs.patch

```diff
@@ -17,3 +35,33 @@
  				Main.tile[i, j].wall = 0;
  
  			WorldGen.UpdateMapTile(i, j);
+@@ -350,18 +_,14 @@
+ 			int num2 = 0;
+ 			if (Main.wallLargeFrames[tile.wall] == 1) {
+ 				num2 = phlebasTileFrameNumberLookup[j % 4][i % 3] - 1;
+-				tile.wallFrameNumber((byte)num2);
+ 			}
+ 			else if (Main.wallLargeFrames[tile.wall] == 2) {
+ 				num2 = lazureTileFrameNumberLookup[i % 2][j % 2] - 1;
+-				tile.wallFrameNumber((byte)num2);
+ 			}
+ 			else if (resetFrame) {
+ 				num2 = WorldGen.genRand.Next(0, 3);
+ 				if (tile.wall == 21 && WorldGen.genRand.Next(2) == 0)
+ 					num2 = 2;
+-
+-				tile.wallFrameNumber((byte)num2);
+ 			}
+ 			else {
+ 				num2 = tile.wallFrameNumber();
+@@ -370,6 +_,10 @@
+ 			if (num == 15)
+ 				num += centerWallFrameLookup[i % 3][j % 3];
+ 
++			if (!WallLoader.WallFrame(i, j, tile.wall, resetFrame, ref num, ref num2))
```

Comment:
should this go before the `centerWallFrameLookup`?

Reply:
I don't think the hook should return, because vanilla provides no way to early return on wall framing

Reply:
The return is so if a user is implementing their own lookup, they can avoid the vanilla lookup being applied. 

centerWallFrameLookup changes the style, so I figure we should provide the chosen style to the modder so they can act on it reliably.